Warning: Multiple assignment operators specified for SpinLockMutex

**Describe your environment** Describe any aspect of your environment relevant to the problem, including your platform, build system, version numbers of installed dependencies, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main branch.

Issue exists in the main branch too, when building we see this warning pop up on using otlp metrics related code;
```
17>c:\work\ic3\d_nsshe_lcsminedgetmp\en-us\amd64chk\packages\opentelemetry.cpp.geneva.1.0.2300\api\include\opentelemetry\common\spin_lock_mutex.h(132) : error C4522: 'opentelemetry::v1::common::SpinLockMutex': multiple assignment operators specified
```

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.
Build with this sample code:
```
#include "opentelemetry/exporters/etw/etw_tracer_exporter.h"
#include "opentelemetry/exporters/otlp/otlp_grpc_metric_exporter_factory.h"
#include "opentelemetry/metrics/provider.h"
#include "opentelemetry/sdk/metrics/export/periodic_exporting_metric_reader.h"
#include "opentelemetry/sdk/metrics/meter.h"
#include "opentelemetry/sdk/metrics/meter_provider.h"

using namespace OPENTELEMETRY_NAMESPACE;
using namespace opentelemetry::exporter::etw;
namespace otlp_exporter = opentelemetry::exporter::otlp;
namespace metrics_sdk = opentelemetry::sdk::metrics;
namespace metrics_api = opentelemetry::metrics;
namespace resource_ns = opentelemetry::sdk::resource;

    void InitOTelMetrics() 
    {
        // Initialize exporter
        otlp_exporter::OtlpGrpcMetricExporterOptions options;
        options.aggregation_temporality = static_cast<opentelemetry::v1::exporter::otlp::PreferredAggregationTemporality>(opentelemetry::v1::sdk::metrics::AggregationTemporality::kDelta);
        options.endpoint = "http://localhost:4317";
        auto exporter = otlp_exporter::OtlpGrpcMetricExporterFactory::Create(options);

        // Initialize metric reader
        metrics_sdk::PeriodicExportingMetricReaderOptions reader_options;
        reader_options.export_interval_millis = std::chrono::milliseconds(60 * 1000); // 1 min
        reader_options.export_timeout_millis  = std::chrono::milliseconds(500);
        std::unique_ptr<metrics_sdk::MetricReader> reader{
            new metrics_sdk::PeriodicExportingMetricReader(std::move(exporter), reader_options)};

        // Initialize MeterProvider
        resource_ns::ResourceAttributes attributes = {
            {"service", ServiceName}, {"_microsoft_metrics_account", "account"}, {"_microsoft_metrics_namespace", "counter"}
        };
        auto resource = resource_ns::Resource::Create(attributes);
        auto views = std::unique_ptr<metrics_sdk::ViewRegistry>(new metrics_sdk::ViewRegistry());
        auto provider = std::shared_ptr<metrics_api::MeterProvider>(new metrics_sdk::MeterProvider(std::move(views), resource));
        auto p = std::static_pointer_cast<metrics_sdk::MeterProvider>(provider);
        p->AddMetricReader(std::move(reader));
        metrics_api::Provider::SetMeterProvider(provider);
    }
```

**What is the expected behavior?**
No such warning is thrown

**What is the actual behavior?**
Warning C4522 is thrown.

**Additional context**
Add any other context about the problem here.

--- a/api/include/opentelemetry/common/spin_lock_mutex.h
+++ b/api/include/opentelemetry/common/spin_lock_mutex.h
@@ -55,7 +55,6 @@ public:
   SpinLockMutex() noexcept {}
   ~SpinLockMutex() noexcept            = default;
   SpinLockMutex(const SpinLockMutex &) = delete;
-  SpinLockMutex &operator=(const SpinLockMutex &) = delete;
   SpinLockMutex &operator=(const SpinLockMutex &) volatile = delete;
 
   static inline void fast_yield() noexcept

--- a/api/include/opentelemetry/common/spin_lock_mutex.h
+++ b/api/include/opentelemetry/common/spin_lock_mutex.h
@@ -55,7 +55,7 @@ public:
   SpinLockMutex() noexcept {}
   ~SpinLockMutex() noexcept            = default;
   SpinLockMutex(const SpinLockMutex &) = delete;
-  SpinLockMutex &operator=(const SpinLockMutex &) volatile = delete;
+  SpinLockMutex &operator=(const SpinLockMutex &) = delete;
 
   static inline void fast_yield() noexcept
   {

